app/service/app_svc.py

```diff
@@ -141,7 +141,11 @@ async def register_contacts(self):
 
     async def validate_requirement(self, requirement, params):
         if not self.check_requirement(params):
-            self.log.error('%s does not meet the minimum version of %s' % (requirement, params['version']))
+            msg = '%s does not meet the minimum version of %s' % (requirement, params['version'])
+            if params.get('optional', False):
+                self.log.warning(msg)
```

Comment:
We should add a bit more context to the message. Something like: "x is an optional dependency and its absence will not affect Caldera's core operation, however some extended features may be impacted."

Alternatively you could add a "reason" field to the requirement definition: 

```yaml
  metasploit:
    command: msfconsole -v
    optional: true
    type: installed_program
    version: 4.0
    reason: Metasploit is an optional dependency and is only required if you would like to launch metasploit modules from Caldera. Caldera's core operation will not be affected if Metasploit is not installed.
```

Then, I think you could just change this line to: 

```python
self.log.warning('%s: %s' % (msg, params.get('reason', ''))
```